### DISCUSSION TITLE
Reusable component composite action python version

### DISCUSSION THREAD

**Discussion by kemayo** [community/community#25457] [Actions]
<p>I have a composite action that I’d like to package up so I can reuse it. It’s just running a python script:</p>
<pre><code class="lang-auto">run: python tools/build_changelog.py CHANGELOG.md RECENT_CHANGES.md
</code></pre>
<p>So, I want to bundle it up. However, the script needs to be run under python3, and I can’t tell what the best way to express this is. Should I:</p>
<ul>
<li>Use <code>setup-python</code> to specify the version in my action</li>
<li>Require the calling workflow to have specified the appropriate python version</li>
</ul>
<p>I guess the root of this is that I’m not sure whether setup-python causes changes to the overall runner environment such that an action running it would maybe change the python version for the entire workflow, or whether everything is entirely walled off.</p>

**Comment by airtower-luna** [comment, marked as answer]
<aside class="quote no-group" data-username="kemayo" data-post="1" data-topic="137833">
<div class="title">
<div class="quote-controls"></div>
 kemayo:</div>
<blockquote>
<p>Use <code>setup-python</code> to specify the version in my action</p>
</blockquote>
</aside>
<p>That’s not possible, composite actions don’t support embedding other options (AFAIK it’s planned, but not implemented). And I assume it would mess with the outside environment anyway.</p>
<p>If you’re okay with limiting the action to current Linux distros you could just call <code>python3</code> instead of <code>python</code>, which should give you the system version.</p>
<p>Or you could make a Docker action, with a suitable Python base image. For a simple script I’d probably use <code>python:3-alpine</code>. That way you can select your own Python environment without affecting the rest of the build.</p>

**Comment by kemayo** [comment]
<aside class="quote no-group" data-username="airtower-luna" data-post="2" data-topic="137833">
<div class="title">
<div class="quote-controls"></div>
 airtower-luna:</div>
<blockquote>
<p>Or you could make a Docker action, with a suitable Python base image. For a simple script I’d probably use <code>python:3-alpine</code> . That way you can select your own Python environment without affecting the rest of the build.</p>
</blockquote>
</aside>
<p>That seems so ridiculously overkill for running a tiny Python script, but I guess if it’s the only way…</p>
<p>Alternately, I guess I could rewrite it to be 2/3 compatible Python and just trust that <code>python</code> will exist and work. I don’t think I’m doing anything that’s going to be too tricky there. :thinking:</p>
